### ISSUE TITLE
[Bug]: Basename is duplicated when calling `setSearchParams`

### ISSUE THREAD

**Issue by ms-tng** [remix-run/react-router#10431]
### What version of React Router are you using?

6.11.0

### Steps to Reproduce

1. Start a React app using this code:
```typescript
import { useEffect } from "react";
import ReactDOM from "react-dom/client";
import { Route, RouterProvider, createBrowserRouter, createRoutesFromElements, useSearchParams } from "react-router-dom";

const Foo = () => {
  const [_, setSearchParams] = useSearchParams();

  useEffect(() => {
    setSearchParams({
      bar: "baz",
    });
  }, [setSearchParams]);

  return null;
};

const router = createBrowserRouter(
  createRoutesFromElements(<Route path="/foo" element={<Foo />} />),
  { basename: "/base" }
);

ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
  <RouterProvider router={router} />
);
```
2. Configure your server to serve the app from `/base`, e.g. with this Vite config:
```typescript
import react from "@vitejs/plugin-react";
import { defineConfig } from "vite";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  server: {
    base: "/base",
  },
});
```
3. Open `/base/foo` in your browser, e.g. with the default Vite port: http://127.0.0.1:5173/base/foo

### Expected Behavior

The app should redirect to http://127.0.0.1:5173/base/foo?bar=baz

This is what happens with `react-router-dom@6.10.0`.

### Actual Behavior

The app redirects to http://127.0.0.1:5173/base/base/foo?bar=baz

I.e. the `/base` part is duplicated. In React's strict mode, when everything is rendered twice in development, it even redirects to http://127.0.0.1:5173/base/base/base/foo?bar=baz

**Comment by brophdawg11** [comment]
Thanks for the reproduction @ms-tng!  Looks like we missed a `stripBasename` call when we moved the relative routing/basename handling inside the `@remix-run/router`

**Comment by brophdawg11** [comment]
This is fixed by #10433 and should be available once 6.11.1 is released

**Comment by ms-tng** [comment]
Works perfectly, thank you for the quick fix! 👍
